src/service/blockchain-event-listener-service.js

```diff
@@ -597,11 +688,47 @@ class BlockchainEventListenerService {
                     hashFunctionId,
                     state,
                     stateIndex,
+                    r0,
                 ),
             ]);
         }
     }
 
+    async handleProofSubmittedEvent(blockEvents) {
```

Comment:
It looks overly complicated to me to handle these additional events just for keeping the bid value up to date. Wouldn't it be simpler to calculate the bid "on the fly" in epoch check command, before scheduling asset sync command the same way it is calculated before storing an asset ?

Reply:
This means bid will need to be calculated only for assets for which rank < r0, and only if asset is not already present in triple store.
It seems like this solution would ensure a lower number of RPC calls and also lower number of mysql calls

Reply:
Listening to events is much less of a load for RPC as far as I understand, that was the original reason for implementing it in this way, also then we don't need to get agreement data from the blockchain for the bid suggestion nor for bid validation on publish. And from what I see we can significantly improve how we're listening to events, right now we're getting logs for every event separately (15 events) which makes no sense as it can be done in 1 call for all contracts/events

Reply:
I think you're trying to over engineer and optimize prematurely here and it can cause issues in the future. If this is supposed to be a "simple asset sync" it would make much more sense imo to start with implementing the feature the simple way, getting agreement data as we do with the publish, and then if necessary optimize the bid calculation in a separate release. 
There are a lot of unknown issues that can come from this. You're making the proper functioning of the protocols dependent on the event listener (what if event listener is lagging behind for some reason and when handling update command you don't have the correct data) ?
We were doing getting events in 1 call for each contract before, but we changed it as for most contracts we only needed a subset of events. Making one call per contract was "heavier" on the rpc from what i remember, and they used to time out.

Reply:
I wouldn't say it's a premature optimization, we've already seen problems with getting data from RPC when it's under big load, it just returns zero values so it's questionable what is more reliable. Even though I don't like the idea of getting SA data from the blockchain, I don't like listening to events even more, so I guess makes sense to optimize it in a separate release